packages/grafana-ui/src/types/datasource.ts

```diff
@@ -208,6 +203,8 @@ export abstract class DataSourceApi<
    * Used by alerting to check if query contains template variables
    */
   targetContainsTemplate?(query: TQuery): boolean;
+
+  supportsStreaming?: boolean;
```

Comment:
Not sure this is the correct way to have this prop. I would appreciate any feedback you have regarding this.

Reply:
I think that was added as a hack to get streaming working for loki... I think it should be removed, or we should be clear about what it means.

The behavior of streaming or not should be controlled by TQuery

Reply:
Do we need it?  If it is used as a high level indication of streaming support that the UI may do something with, maybe we should put it in [DataSourcePluginMeta](https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/types/datasource.ts#L77) with the rest of the properties like that

Reply:
It's mainly used by the RefreshPicker to add an `Live` option, it could be removed from all other places. DataSourcePluginMeta sounds like a better fit. I'll look into that. Thanks